liblwgeom/lwgeom_geos.c

```diff
@@ -112,7 +112,7 @@ GEOS2LWGEOM(const GEOSGeometry *geom, char want3d)
 {
 	int type = GEOSGeomTypeId(geom) ;
 	int hasZ;
-	int SRID = GEOSGetSRID(geom);
+	int SRID = (int32_t) GEOSGetSRID(geom);
```

Comment:
This one looks strange.
Gets an int, casts to 32bit, casts to int.

Reply:
Yes. I'll reset this file to its current status and review it again